probe/data.go

```diff
@@ -95,8 +97,41 @@ func LoadDataFromFile(filename string) error {
 		return err
 	}
 
-	time := time.Now().Format(time.RFC3339)
+	time := time.Now().UTC().Format(time.RFC3339)
 	os.Rename(filename, filename+"-"+time)
 
 	return nil
 }
+
+// CleanDataFile keeps the max backup of data file
+func CleanDataFile(filename string, backups int) {
+
+	// if backups is negative value, keep all backup files
+	if backups < 0 {
+		return
+	}
+
+	// get all of the backup files
+	pattern := filename + "-*"
+	matches, err := filepath.Glob(pattern)
+	if err != nil {
+		log.Errorf("Cannot clean data file: %v", err)
+		return
+	}
+
+	// if backups is not exceed the max number of backup files, return
+	if len(matches) <= backups {
+		log.Debugf("No need to clean data file (%d - %d) ", backups, len(matches))
+	}
+
+	// remove the oldest backup files
+	sort.Strings(matches)
+
+	for i := 0; i < len(matches)-backups; i++ {
```

Comment:
If I understand correctly the negative value will lead to removing all backup files? The behavior is deviating from the description of the PR.

Reply:
At the beginning of this function, we checked the negative value. it would keep all backup files.

```go
	// if backups is negative value, keep all backup files
	if backups < 0 {
		return
	}
```